modules/p2/ting_openlist/ting_openlist.module

```diff
@@ -240,128 +242,74 @@ function ting_openlist_field_validate($element, &$form_state, $form) {
- *   active.
  *
  * @return mixed
  *   Response from Openlist.
  */
-function ting_openlist_request($function, array $arguments = array(), $detailed_log = FALSE) {
-  $response = FALSE;
-  $start_time = microtime(TRUE);
-
-  if ($soap_client = ting_openlist_get_soap()) {
-    try {
-      $response = call_user_func_array(
-        array(
-          $soap_client,
-          $function,
-        ),
-        $arguments
-      );
-
-      // If logging is enabled, log the function name and execution time.
-      if (variable_get('ting_openlist_log', FALSE)) {
-        $execution_time = microtime(TRUE) - $start_time;
-
-        $details = '';
-        if ($detailed_log || variable_get('ting_openlist_log_details', FALSE)) {
-          $details = "<pre>" . var_export($arguments, TRUE) . '</pre>';
-        }
+function ting_openlist_request($function, array $arguments = array()) {
+  timer_start(__FUNCTION__);
 
-        watchdog(
-          'openlist',
-          '@function (@execution_time s)!details',
-          array(
-            '@function' => $function,
-            '@execution_time' => $execution_time,
-            '!details' => $details,
-          ),
-          WATCHDOG_INFO
-        );
-      }
-    }
-    catch (Exception $e) {
-      $execution_time = microtime(TRUE) - $start_time;
-      watchdog(
-        'openlist',
-        'Error (@execution_time s).<br /><strong>Function:</strong> @function.<br /><strong>Exception:</strong>@exception<hr /><pre>!details</pre>',
-        array(
-          '@function' => $function,
-          '@execution_time' => $execution_time,
-          '@exception' => $e->getMessage(),
-          '!details' => var_export($arguments, TRUE),
-        ),
-        WATCHDOG_ERROR
-      );
-
-      throw $e;
-    }
+  $parts = array(
+    variable_get('ting_openlist_domain', TING_OPENLIST_DEFAULT_DOMAIN),
+    'json',
+    $function,
+  );
+
+  $query = array(
+    'library_code' => variable_get('ting_library_code', ''),
+    'authkey' => variable_get('ting_openlist_authkey', ''),
+  );
+
+  if (variable_get('ting_openlist_developer', FALSE)) {
+    $query['developer'] = 'on';
   }
-  else {
-    throw new Exception('No soap client available.');
+
+  $query += $arguments;
+
+  $url = url(
+    implode('/', $parts),
+    array('query' => $query)
+  );
+
+  $response = drupal_http_request($url);
+  $data = drupal_json_decode($response->data);
+
+  if ($response->code != '200' || empty($data)) {
+    watchdog(
+      'ting_openlist',
+      '<strong>Request</strong><pre>!link</pre><strong>Execution time</strong><pre>@execution_time ms</pre><strong>Response</strong><pre>!response</pre>',
+      array(
+        '!link' => l($url, $url, array('external' => TRUE)),
+        '@execution_time' => timer_read(__FUNCTION__),
+        '!response' => htmlentities(var_export($response, TRUE)),
+      ),
+      WATCHDOG_ERROR
+    );
+
+    throw new TingOpenlistRequestException($response);
```

Comment:
Now that we have a proper exception to work with remember to update all the openlist related code which previously caught the generic `Exception` to catch this class instead.

[PHP code may throw exceptions for many different reasons](http://php.net/manual/en/language.errors.php7.php) and catching them all can make problems difficult to track down.